docs/VLSI/Basic-Flow.rst

```diff
@@ -0,0 +1,217 @@
+.. _hammer_basic_flow:
+
+Using Hammer To Place and Route a Custom Block
+=================================================
+
+.. IMPORTANT:: In order to use the Hammer VLSI flow, you need access to Hammer tools and technology plugins. You can obtain these by emailing hammer-plugins-access@lists.berkeley.edu with a request for which plugin(s) you would like access to. 
+
+Initialize the Hammer Plug-ins
+----------------------------------
+In the Chipyard root, run:
+
+.. code-block:: shell
+
+    ./scripts/init-vlsi.sh <tech-plugin-name>
+    
+This will pull the Hammer & CAD tool plugin submodules, assuming the technology plugins are available on github.
```

Comment:
The tech plugin doesn't have to be in github. If it's not one of the default plugins in Hammer (asap7, nangate45 <- you need to add this to the vlsi Makefile), then it needs to be added as a submodule in the `vlsi/` folder first before you run the `init-vlsi.sh` script.

Reply:
Also I'm a bit confused about where this fits into the existing docs.

Reply:
I'm not sure I understand this comment. I know that the tech plugin doesn't have to be in github. That's the reason the following lines explain how to clone a private tech plug-in.

Why does the submodule need to be added before `init-vlsi.sh`. The script only initializes the submodules (rescursively). If you're cloning the submodule independently it will already be initialized, wouldn't it? And AFAIK, tech-plugins don't include submodules within them

Reply:
Don't you need to register the submodule by doing a `git submodule add` first?